kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/challenges/flightsearch/FlightSearchEngine.java

```diff
@@ -0,0 +1,53 @@
+package com.kodilla.good.patterns.challenges.flightsearch;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FlightSearchEngine {
+
+    private Set<Flight> listOfFlights;
+
+    public FlightSearchEngine(Set<Flight> listOfFlights) {
+        this.listOfFlights = listOfFlights;
+    }
+
+    public Set<Flight> searchByFlightDeparture(String departure) {
+        System.out.println("Searching all Flights by departure from: " + departure);
+        listOfFlights.stream()
+                .filter(dep -> dep.getDeparture().equals(departure))
+                .forEach(System.out::println);
+        return listOfFlights.stream()
+                .filter(dep -> dep.getArrival().equals(departure))
+                .collect(Collectors.toSet());
+    }
+
+    public Set<Flight> searchByFlightArrival(String arrival) {
+        System.out.println("Searching all Flights by arrival to: " + arrival);
+        listOfFlights.stream()
+                .filter(arr -> arr.getArrival().equals(arrival))
+                .map(string -> string.toString())
+                .forEach(System.out::println);
+        return listOfFlights.stream()
```

Comment:
Nie potrzebujesz dwa razy tworzyć streama. Aby osiągnąć wypisywanie elementów na konsolę (do czego zniechęcam w tym miejscu - patrz inny komentarz), możesz użyć metody `peek` na strumieniu.

Reply:
(opcjonalnie) możesz zrobić metodę prywatną `searchBy` przyjmującą jako argument `Predicate<String>`. Wtedy np. metoda `searchByFlightDeparture` wyglądałaby tak:
```
public List<FlightSearchResult> searchByFlightDeparture(String departure) {
    return searchBy(flight -> flight.getDeparture().equals(departure));
}
```
Gdybyś nie wiedział, jak to działa, to odszukaj proszę w modułach pojęcie _interfejsu funkcjonalnego_.